lib/text/vtt_text_parser.js

```diff
@@ -439,6 +440,77 @@ shaka.text.VttTextParser = class {
     }
   }
 
+  /**
+   * Converts voice style tag to be valid for xml parsing
+   * For example,
+   * input: <v Shaka>Test
+   * output: <v.voice-Shaka>Test</v.voice-Shaka>
+   *
+   * @param {string} payload
+   * @return {string} processed payload
+   * @private
+   */
+  static replaceVoiceStylePayload_(payload) {
+    const voiceTag = 'v';
+    const names = [];
+    let nameStart = -1;
+    let newPayload = '';
+    let hasVoiceEndTag = false;
+    for (let i = 0; i < payload.length; i++) {
+      if (payload[i] === '/') {
```

Comment:
I think this method could use more comments too. I had to spend a while reading this to figure out that this clause is for handling tags that already have an end tag.

Reply:
OK, I'll add some comments!

Reply:
Done!